controller/work_item_link_category_blackbox_test.go

```diff
@@ -142,12 +146,90 @@ func (s *workItemLinkCategorySuite) createWorkItemLinkCategoryUser() (http.Respo
+}
+
+func createWorkItemLinkCategoryInRepo(t *testing.T, db application.DB, ctx context.Context, name string, description string, id *string) uuid.UUID {
+	var cID uuid.UUID
+	if id == nil {
+		cID = uuid.NewV4()
+	} else {
+		cID = uuid.FromStringOrNil(*id)
+	}
+
+	// Use the goa generated code to create a work item link category
+	payload := app.CreateWorkItemLinkCategoryPayload{
+		Data: &app.WorkItemLinkCategoryData{
+			ID:   &cID,
+			Type: link.EndpointWorkItemLinkCategories,
+			Attributes: &app.WorkItemLinkCategoryAttributes{
+				Name:        &name,
+				Description: &description,
+			},
+		},
+	}
+
+	err := application.Transactional(db, func(appl application.Application) error {
+		modelCategory := ConvertLinkCategoryToModel(app.WorkItemLinkCategorySingle{Data: payload.Data})
+		_, err := appl.WorkItemLinkCategories().Create(ctx, &modelCategory)
+		return err
+	})
+	require.Nil(t, err)
+	return cID
+}
+
 //-----------------------------------------------------------------------------
 // Actual tests
 //-----------------------------------------------------------------------------
 
+func (s *workItemLinkCategorySuite) TestCreateAndDeleteWorkItemLinkCategoryFails() {
+	name := "test-user"
+	description := "This work item link category is managed by an admin user."
+	id := uuid.FromStringOrNil("bf30167a-9446-42de-82be-6b3815152051")
+
+	// Use the goa generated code to create a work item link category
+	payload := app.CreateWorkItemLinkCategoryPayload{
+		Data: &app.WorkItemLinkCategoryData{
+			ID:   &id,
+			Type: link.EndpointWorkItemLinkCategories,
+			Attributes: &app.WorkItemLinkCategoryAttributes{
+				Name:        &name,
+				Description: &description,
+			},
+		},
+	}
+
+	test.CreateWorkItemLinkCategoryMethodNotAllowed(s.T(), s.svc.Context, s.svc, s.linkCatCtrl, &payload)
+	test.DeleteWorkItemLinkCategoryMethodNotAllowed(s.T(), s.svc.Context, s.svc, s.linkCatCtrl, id)
+}
+
+func (s *workItemLinkCategorySuite) TestUpdateWorkItemLinkCategoryFails() {
+	name := "Some name"
+	description := "New description for work item link category."
+	id := uuid.FromStringOrNil("88727441-4a21-4b35-aabe-007f8273cd19")
+	payload := &app.UpdateWorkItemLinkCategoryPayload{
+		Data: &app.WorkItemLinkCategoryData{
+			ID:   &id,
+			Type: link.EndpointWorkItemLinkCategories,
+			Attributes: &app.WorkItemLinkCategoryAttributes{
+				Name:        &name,
+				Description: &description,
+			},
+		},
+	}
+	test.UpdateWorkItemLinkCategoryMethodNotAllowed(s.T(), s.svc.Context, s.svc, s.linkCatCtrl, *payload.Data.ID, payload)
+}
+
+// Currently not used. Disabled as part of https://github.com/fabric8-services/fabric8-wit/issues/1299
 // TestCreateWorkItemLinkCategory tests if we can create the "test-system" work item link category
-func (s *workItemLinkCategorySuite) TestCreateAndDeleteWorkItemLinkCategory() {
+func (s *workItemLinkCategorySuite) xTestCreateAndDeleteWorkItemLinkCategory() {
```

Comment:
I'd say we shouldn't use either of those but have `bool` test flag (defined in `controller_test` package) that is disabled by default. If that flag is enabled we skip the test that shall be disabled.

```go
// Define as many of such flags and skip texts somewhere
// in the controller_test package as you feel are needed.
var enableTestsLinkTypeCreation = flag.Bool("enableTestsLinkTypeCreation", false, "when set tests with link type creation are enabled")
const linkTypeCreationSkipText = "skipped because test flag \"-enableTestsLinkTypeCreation\" is not given"

func TestSomethingWithLinkTypeCreation(t *testing.T) {
  if !*enableTestsLinkTypeCreation {
    t.Skip(linkTypeCreationSkipText)
  }
}
```

I think the effort for disabling code is the same but in order to get those tests back, we have almost no work to do.